include/highfive/bits/H5Utils.hpp

```diff
@@ -57,6 +58,12 @@ struct array_dims<T[N]> {
     static const size_t value = 1 + array_dims<T>::value;
 };
 
+// Only supporting 1D arrays at the moment
+template<typename T, std::size_t N>
+struct array_dims<std::array<T,N>> {
+    static const size_t value = 1;
```

Comment:
Shoulnd't it be?

    static const size_t value = 1 + array_dims<T>::value;

Reply:
All the const can be constexpr

Reply:
Change the existing ones too? `static constexpr size_t`?

Reply:
Yes, I'd do it

Reply:
Done.